### ISSUE TITLE
Issues with Tensorflow 2.2 on Google Colab

### ISSUE THREAD

**Issue by Bastouuuu** [charlesq34/pointnet2#194]
Hello,
Tried to run the code on Google Colab but ran into a few issues. 
When using tf 2.2, an error is raised because tf.placeholder doesn't exist anymore in v2.2. 
To solve this, I did the following (in all files importing TensorFlow) : 
`import tensorflow.compat.v1 as tf`
`tf.disable_v2_behavior()`

But it now raises an error in tf_util.py when tf.contrib is called. 
`AttributeError: module 'tensorflow.compat.v1' has no attribute 'contrib'`

Any idea on how to fix this ? (I can't seem to be able to connect to a GPU when downgrading TF to v1.4)

**Comment by charlesq34** [comment]
I think TF2 no longer supports tf.contrib (https://www.tensorflow.org/guide/migrate)

I guess you need to find a way to replace the two tf.contrib.layers used:
tf.contrib.layers.xavier_initializer()
tf.contrib.layers.batch_norm

**Comment by Bastouuuu** [comment]
Yes, that's what I figured. Someone already opened a PR with changes to replace tf.contrib but quite a few parameters are completely ignored that way.
I'll look around and see if I can find a better approach to replace tf.contrib. 
Thanks for the quick answer !

**Comment by Susiehub** [comment]
@Bastouuuu @charlesq34 
I need to use Tensorflow2.x version to be compatible to GPU.
Did any of you get a working alternative for 

tf.contrib.layers.xavier_initializer()
tf.contrib.layers.batch_norm

**Comment by Susiehub** [comment]
The following functions worked for me
1. tf.keras.initializers.glorot_uniform() instead of tf.contrib.layers.xavier_initializer()

 2. tf.layers.batch_normalization(inputs, 
                                      center=True, scale=True,
                                      training=is_training, momentum=bn_decay)

instead of tf.contrib.layers.batch_norm

**Comment by EXJUSTICE** [comment]
@Susiehub how did you manage the custom operator comparison step? Did you do it on your local machine and then add the operators to your repo for colab?
